docs/docs/api/rest.md

```diff
@@ -0,0 +1,8 @@
+# REST API
+
+Mathesar has a REST API that the front end uses to interact with the backend.
+
+For Mathesar's beta release, we are actively transitioning to a new [RPC-style API](https://wiki.mathesar.org/projects/2024/architecture-transition/rpc/) and will soon be phasing out the REST API entirely.
```

Comment:
@mathemancer are we going to be entirely removing the REST API?

Reply:
I'm wondering if it makes sense to have a REST API at some point in the future for other API consumers / apps, etc., but I don't think we need to think about that now.